sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -487,7 +488,8 @@ private[parquet] class ParquetRowConverter(
     parquetType.asPrimitiveType().getPrimitiveTypeName == INT64 &&
       parquetType.getLogicalTypeAnnotation.isInstanceOf[TimestampLogicalTypeAnnotation] &&
       !parquetType.getLogicalTypeAnnotation
-        .asInstanceOf[TimestampLogicalTypeAnnotation].isAdjustedToUTC
+        .asInstanceOf[TimestampLogicalTypeAnnotation].isAdjustedToUTC &&
+      Utils.isTesting
```

Comment:
can we add a TODO to remove this when NTZ is ready?

Reply:
@cloud-fan Since we decided to release NTZ in 3.4 version, there should not be TODO in 3.3 code base.
We can have a simple comment like 
```
// SPARK-38829: Remove TimestampNTZ type support in Parquet for Spark 3.3
```

Reply:
SGTM